### ISSUE TITLE
octo.exe not overriding default package version for package reference in script step when using PackageId:Version format in --package argument

### ISSUE THREAD

**Issue by hnrkndrssn** [OctopusDeploy/Issues#4944]
**_Are you a customer of Octopus Deploy? Don't raise the issue here. Please contact [our support team](https://octopus.com/support) so we can triage your issue, making sure it's handled appropriately._**

# Prerequisites

- [x] I have verified the problem exists in the latest version
- [x] I have searched [open](https://github.com/OctopusDeploy/Issues/issues) and [closed](https://github.com/OctopusDeploy/Issues/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed) issues to make sure it isn't already reported
- [x] I have written a descriptive issue title
- [x] I have linked the original source of this report
- [x] I have tagged the issue appropriately (area/*, kind/bug, tag/regression?)

# The bug

If you have a script step that references a package and you use the `--package <PackageId>:<Version>` to override the version used in `--defaultpackageversion` it will not select the correct version (it will use the version specified in `--defaultpackageversion`).

## What I expected to happen

I expect it to use the version specified in `--package <PackageId>:<Version>`.

### Log exerpt

```
& $octo create-release --project "MyProject" --server $server --apiKey $apiKey --releaseNumber "1.0.0" --defaultpackageversion "1.0.0" --package "ThePackage:1.0.1"
Octopus Deploy Command Line Tool, version 4.41.3
...
  #   Name               Version       Source           Version rules      
  --- ------------------ ------------- ---------------- -------------------
  1   MyPackage           1.0.0   User specified   Allow any version  
  2   ThePackage          1.0.1       User specified   Allow any version  
```

## Workarounds

Use `--package <StepNameOrAsterisk>:<PackageId>:<Version>` instead.

## Links

https://secure.helpscout.net/conversation/677207670/33058?folderId=557077

**Comment by graemechristie** [comment]
This looks to be fixed. At least we were hitting this issue. After updating our Octopus Client to "Octopus Deploy Command Line Tool, version 6.3.3" (in this case being run via the team city plugin), the issue no longer occurs.

**Comment by slewis74** [comment]
Closing this, as it's either already fixed or will be covered by #5821
